ext/standard/array.c

```diff
@@ -6549,32 +6549,25 @@ PHP_FUNCTION(array_filter)
 				if (!string_key) {
 					ZVAL_LONG(key, num_key);
 				} else {
-					ZVAL_STR_COPY(key, string_key);
+					ZVAL_STR(key, string_key);
 				}
 			}
 			if (use_type != ARRAY_FILTER_USE_KEY) {
-				ZVAL_COPY(&args[0], operand);
+				ZVAL_COPY_VALUE(&args[0], operand);
 			}
 			fci.params = args;
```

Comment:
Somewhat surprisingly, moving this line out of the loop into the `ZEND_FCI_INITIALIZED(fci)` decreases performance. I don't have an explanation for that.